### ISSUE TITLE
Discard cache after resolution

### ISSUE THREAD

**Issue by fregante** [sindresorhus/p-memoize#20]
Like `cachePromiseRejection`, but instead of deleting on errors, it deletes itself on resolution.

More details: https://github.com/sindresorhus/promise-fun/issues/15#issuecomment-871307098

If accepted, this would be trivial to implement.

**Comment by sindresorhus** [comment]
Good idea 👍

**Comment by etienne-martin** [comment]
Passing `maxAge: 0` seems to yield the same behavior.

**Comment by fregante** [comment]
I haven’t tested by I think that just kills the memoization entirely. That option is not promise-aware so the timer starts when the memoized function is called and then the cached promise is immediately removed in the next timeout.

**Comment by etienne-martin** [comment]
You're right, the reason it worked is that I was firing multiple invocation at the same time. 

Here's [a demo](https://codesandbox.io/s/wizardly-hoover-vxrxu?file=/src/index.js), we can see that the first two invocations are memoized but the 3rd one (which is fired while the other two promises are pending) didn't resolve from the cache.

**Comment by Richienb** [comment]
In the context of #26: Don't actaully set anything to `cache` and remove promises from `promiseCache` when they finish executing (probably in a `finally` statement after the try/catch block).

**Comment by Richienb** [comment]
You can now achieve this by making `cache` a black hole.

**Comment by Richienb** [comment]
`cachePromiseRejection` was removed because we recognise that the errors can't be serialised well. If the consumer uses a database to store results, there's no point enforcing our own standard for caching errors.

**Comment by fregante** [comment]
Oh you mean that the cache itself will track the resolution and remove anything that’s resolved?

**Comment by Richienb** [comment]
The trick is just to have the result cache not cache anything

**Comment by fregante** [comment]
I think I had misunderstood the change in v7. This change request is actually rather simple:

https://github.com/sindresorhus/p-memoize/blob/f469c0616864258c7e31fc3636e0f222340a1ada/index.ts#L115

``` js 
if (!onlyWhilePending) {
  cache.set()
}
```

but I agree that passing a map without .set also works, except it’d look rather weird without knowing pMemoize’s internal behavior.

**Comment by Richienb** [comment]
A more elegant solution could be to allow `cache: false`.

**Comment by fregante** [comment]
I like that. Still unreadable but better than a new Map type

**Comment by fregante** [comment]
Thank you!
